### PR TITLE
Sami Peterson Mid Mod Refactor

### PR DESCRIPTION
**Pull request by sami-p**
The only concepts that are still a bit unclear to me are including an image for each member, and using within blocks to test more thoroughly. I would also love some feedback on how to properly test the service and facade, or anything else that you notice I should have tested! Thank you!

### CODE REVIEW

File: app/controllers/search_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class SearchController < ApplicationController
+  def index
+    nation = params[:nation]
+    @nation = nation.gsub!("_", " ")
```
Comment (megstang) [inline app/controllers/search_controller.rb:4]: 👍

File: app/facades/member_facade.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class MemberFacade
+  def self.search(nation)
+    member_data = AirBenderService.get_members(nation)
```
Comment (megstang) [inline app/facades/member_facade.rb:3]: I may consider renaming this `get_members` to something a bit more related to what that service method will be doing -- `members_in_nation` (?)

File: app/poros/member.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+              :enemies,
+              :affiliation
+
+  def initialize(data)
+    @name = data[:name]
+    @allies = data[:allies]
+    @enemies = data[:enemies]
+    @affiliation = data[:affiliation]
+  end
+end
```
[the remark this answers is not in the document]
Comment (megstang) [reply app/poros/member.rb:13]: To display an image an an `html.erb` file, specifically one that comes from a url (like this one), you can use an image_tag (helper that comes from ruby). Here are some docs on that image_tag: https://apidock.com/rails/ActionView/Helpers/AssetTagHelper/image_tag


TLDR: <%= image_tag 'https://your-url-for-photo-here.com' %>

File: app/services/air_bender_service.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class AirBenderService
+  def self.get_members(nation)
+    response = Faraday.get("https://last-airbender-api.herokuapp.com/api/v1/characters?perPage=150&affiliation=#{nation}")
+    JSON.parse(response.body, symbolize_names: true)
```
Comment (megstang) [inline app/services/air_bender_service.rb:4]: Nice job here. Since the github docs for this api mention that there are 497 members, and no other way to specify that you want ALL of the characters back for a particular nation, I think I'd just set the perPage to 497 to ensure that I get ALL of the characters back for a particular nation (because, what if a particular nation had 151 characters in it?)

File: app/views/search/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<h1><%= @nation.titleize %></h1>
+
+<h2>Nation Population: <%= @members.count %></h2>
+<h2>First 25 Members:</h2>
+<ol><% @members[0..24].each do |member| %>
```
Comment (megstang) [inline app/views/search/index.html.erb:5]: It would probably be best, optimization wise, if the members array was limited before getting to the view -- @ members contains 97 (or so) member objects, and you only ever want to show 25 on this index page, so your facade is doing a lot of extra work creating so many objects that you don't ever want to show on the page. I'll leave a comment back on your facade with a way to limit the members there.

File: app/facades/member_facade.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+    member_data.map do |member|
+      Member.new(member)
+    end
+  end
```
Comment (megstang) [inline app/facades/member_facade.rb:7]: We want both total member count AND 25 member objects to get sent back to the view. 

You could do something like this: 

```ruby 
def self.search(nation)
    member_data = AirBenderService.get_members(nation)
    twenty_five_members = member_data[0..24].map do |member|
      Member.new(member)
    end
    return [member_data.count, twenty_five_members]
end 
```

Then, in your controller, you could assign both return values to variables by doing something like this: 

```ruby
@member_count, @members = MemberFacade.search(nation)
```

File: app/views/search/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+              <% end %>
+            <% else %> N/A
+            <% end %><br>
+    Enemies: <% if member.enemies.count > 0 %>
+              <% member.enemies.each do |enemy| %>
+                <%= enemy %>
+              <% end %>
+            <% else %> N/A
+            <% end %><br>
+    Affiliation: <%= member.affiliation %></p></li>
```
Comment (megstang) [inline app/views/search/index.html.erb:20]: If you wanted to use dynamic id's you could do something like this: 
```html.erb
<div id="member-#{member.id}">
<li><p>Name: <%= member.name %>
  <br>
    Allies: <% if member.allies.count > 0 %>
              <% member.allies.each do |ally| %>
                <%= ally %>
              <% end %>
            <% else %> N/A
            <% end %><br>
    Enemies: <% if member.enemies.count > 0 %>
              <% member.enemies.each do |enemy| %>
                <%= enemy %>
              <% end %>
            <% else %> N/A
            <% end %><br>
    Affiliation: <%= member.affiliation %></p></li>
</div>
```

**Review by megstang — COMMENTED** [review]
Let me know if you have any questions on these comments! I'll follow up in a DM with more feedback on testing.
